src/Sulu/Bundle/Sales/CoreBundle/Transition/TransitionResolver.php

```diff
@@ -0,0 +1,241 @@
+        /** @var Transition $transition */
+        $transition = $this->transitionManager->findOneByDestination($alias, $id);
+
+        $transitionResult = $this->createTransitionResult($alias, $id, $hydrationMode);
+
+        $transitionResult->setItems($transition->getItems()->toArray());
+        $transitionResult->setId($transition->getId());
+        $transitionResult->setTransition($transition);
+
+        return $transitionResult;
+    }
+
+    /**
+     * @param string $alias
+     * @param int $id
+     * @param string $hydrationMode
+     *
+     * @return TransitionResultInterface[]
+     */
+    protected function getPreviousTransitions($alias, $id, $hydrationMode = self::HYDRATION_MODE_OBJECTS)
+    {
+        $transitionResults = [];
+        $transition = $this->transitionManager->findOneByDestination($alias, $id);
+
+        if ($transition != null) {
+            $transitionResult = $this->createTransitionResult(
+                $transition->getDestination(),
+                $transition->getDestinationId(),
+                $hydrationMode
+            );
+
+            $transitionResult->setItems($transition->getItems()->toArray());
+            $transitionResult->setId($transition->getId());
+            $transitionResult->setTransition($transition);
+
+            $transitionResults[] = $transitionResult;
+
+            $currentAlias = $transition->getSource();
+            $currentId = $transition->getSourceId();
+
+            // recursive call to get all previous of previous....
+            while ($previousTransitions = $this->getPreviousTransitions($currentAlias, $currentId, $hydrationMode)) {
+                array_merge($transitionResults, $previousTransitions);
+                $currentAlias = $previousTransitions[0]->getTransition()->getSource();
+                $currentId = $previousTransitions[0]->getTransition()->getSourceId();
+            }
+
+            $firstTransitionResult = $this->createTransitionResult($currentAlias, $currentId, $hydrationMode);
+
+            $transitionResults[] = $firstTransitionResult;
+
+        }
+        $transitionResults = array_reverse($transitionResults);
+
+        return $transitionResults;
+    }
+
+    /**
+     * @param $alias
+     * @param $id
+     * @param $hydrationMode
+     *
+     * @return TransitionResult
+     */
+    protected function createTransitionResult($alias, $id, $hydrationMode)
+    {
+        $parameters = $this->dependencyManager->getParametersForAlias($alias);
+        $transitionResult = new TransitionResult();
+
+        $this->setParameters($transitionResult, $parameters, $id);
+
+        $number = null;
+        $created = null;
+        if ($hydrationMode == self::HYDRATION_MODE_OBJECTS) {
+            /** @var EntityRepository $entityRepository */
+            $entityRepository = $this->entityManager->getRepository($parameters['class']);
+            $entity = $entityRepository->find($id);
+
+            $number = $entity->getNumber();
```

Comment:
what if functions are not set on class?
either use `method_exists(..)`
or (even better) define an interface and check for that interface